pages/qr.tsx

```diff
@@ -5,10 +5,30 @@ import { Text, Card, Box } from '~ui/core'
 import { isFormal } from '~lib/config'
 import { ArrowsLeft, ArrowsRight } from '~ui/anicons'
 import { MobileApp } from '~ui/layouts/MobileApp'
+import { useRouter } from 'next/router'
 
 export default function QRCodePage() {
   const videoEl = React.useRef<HTMLVideoElement>()
 
+  const router = useRouter()
+
+  const paramsRef = React.useRef<object>({})
+
+  paramsRef.current = {
+    name: router.query.name?.toString(),
+    phone: router.query.phone?.toString(),
+    address: router.query.address?.toString(),
+  }
+
+  function appendGuestInfo(url: string): any {
+    Object.entries(paramsRef.current).forEach((entry) => {
+      if (entry[1]) {
+        url = url.concat(`&${entry[0]}=${entry[1]}`)
+      }
+    })
+    return url
+  }
```

Comment:
How about using the URL API? [URLSearchParams](https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams)

Reply:
Nice API, I will check it out.